CodePush.ios.js

```diff
@@ -210,7 +209,13 @@ function sync(options = {}, syncStatusChangeCallback, downloadProgressCallback)
           resolve(CodePush.SyncStatus.UP_TO_DATE);
         }
         else if (syncOptions.updateDialog) {
-          syncOptions.updateDialog = Object.assign(CodePush.DEFAULT_UPDATE_DIALOG, syncOptions.updateDialog);
+          // updateDialog supports any truthy value (e.g. true, "goo", 12),
+          // but we should treat a non-object value as just the default dialog
+          if (typeof syncOptions.updateDialog !== "object") {
+            syncOptions.updateDialog = CodePush.DEFAULT_UPDATE_DIALOG;
+          } else {
+            syncOptions.updateDialog = Object.assign(CodePush.DEFAULT_UPDATE_DIALOG, syncOptions.updateDialog);
```

Comment:
To ensure DEFAULT_UPDATE_DIALOG doesn't get modified, we should do:
```
syncOptions.updateDialog = Object.assign({}, CodePush.DEFAULT_UPDATE_DIALOG, syncOptions.updateDialog);
```

And I am pretty sure that the typecheck is not necessary. If the source (syncOptions.updateDialog) does not contain any properties, i.e it is an integer or a boolean, it will just be ignored.